cupyd/core/etl.py

```diff
@@ -237,9 +239,8 @@ def _build(self, num_workers: int, monitor_performance: bool):
                 )
 
         # 5. Create necessary structures & ETL Workers
-        worker_num = 1
-        finished_workers = Queue()
-        node_timings = Queue(maxsize=100_000)  # todo: might cause slowdowns, rethink?
+        finished_workers: Queue[Tuple[str, str, Dict[str, NodeException]]] = Queue()
+        node_timings: Queue[Tuple[str, float]] = Queue(maxsize=25_000)
```

Comment:
limited for 25k to fix the macOS issue. Since we might rewrite how to handle node timings, I just set this value to be high but at least lower than the macOS semaphore size limit.